server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -336,8 +336,8 @@ public DesiredBalance compute(
             assignments.put(shardAndAssignments.getKey(), ShardAssignment.ofAssignedShards(shardAndAssignments.getValue()));
         }
 
-        for (var ignored : routingNodes.unassigned().ignored()) {
-            var info = ignored.unassignedInfo();
+        for (var shard : routingNodes.unassigned().ignored()) {
```

Comment:
Just to check I'm not missing anything here - we're just renaming `ignored -> shard` and extracting `unassigned ? 0 : 1` as `ignored`, not changing any behaviour right?